docs/2022-07-18-JSON-RPC-Server-API.md

```diff
@@ -87,52 +87,106 @@ If the verification of the `state` pattern fails, the following error is returne
 
 ### Correct Response:
 
+All correct responses have a result containing a `reason`, with one of the following values:
+* `"reason: "branching"`
+* `"reason: "stuck"`
+* `"reason: "depth-bound"`
+* `"reason: "cut-point-rule"`
+* `"reason: "terminal-rule"`
+
+A field `state` contains the state reached (including optional `predicate` and `substitution`), a field `depth` indicates the execution depth.
+
 ```json
 {
   "jsonrpc": "2.0",
   "id": 1,
   "result": {
-    "states": [
-      {
-        "state": {"format":"KORE", "version":1, "term":{}},
-        "depth": 1
-      }
-    ],
-    "reason": "final-state"
+    "state": {
+      "state": {"format":"KORE", "version":1, "term":{}},
+      "predicate": {"format":"KORE", "version":1, "term":{}},
+      "substitution": {"format":"KORE", "version":1, "term":{}},
+    },
```

Comment:
There is more work to do here... a `Substitution` is not equivalent to a `KorePattern` (rather a collection of pairs of `KorePattern`, or a map from `Text` to `KorePattern`)